pkgs/build-support/dotnet/make-nuget-deps/default.nix

```diff
@@ -1,14 +1,30 @@
-{ linkFarmFromDrvs, fetchurl }:
+{ linkFarmFromDrvs, substituteAll, fetchurl, stdenvNoCC, zip }:
 { name, nugetDeps ? import sourceFile, sourceFile ? null }:
 linkFarmFromDrvs "${name}-nuget-deps" (nugetDeps {
   fetchNuGet = { pname, version, sha256 ? "", hash ? ""
     , url ? "https://www.nuget.org/api/v2/package/${pname}/${version}" }:
-    fetchurl {
+    stdenvNoCC.mkDerivation {
       name = "${pname}.${version}.nupkg";
-      # There is no need to verify whether both sha256 and hash are
-      # valid here, because nuget-to-nix does not generate a deps.nix
-      # containing both.
-      inherit url sha256 hash;
+      src = fetchurl {
+        name = "${pname}.${version}.nupkg.${url}";
+        # There is no need to verify whether both sha256 and hash are
+        # valid here, because nuget-to-nix does not generate a deps.nix
+        # containing both.
+        inherit url sha256 hash;
+      };
+      dontUnpack = true;
+      nativeBuildInputs = [ zip ];
+      installPhase = "cp $src $out";
+      # NuGet.org edits packages by signing them during upload, which makes
+      # those packages nondeterministic. Strip out the signature file.
+      # Signing logic is https://github.com/NuGet/NuGet.Client/blob/128a5066b1438627ac69a2ffe9de564b2c09ee4d/src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveIOUtility.cs#L518
+      # Non-NuGet.org sources might not have a signature file.
+      fixupPhase = ''
+        tmp_file=$(mktemp tmp-XXXXXXXX.zip)
+        cp "$out" "$tmp_file"
+        zip -d "$tmp_file" .signature.p7s || true
```

Comment:
what sort of error are we ignoring here?  just cases where there's no signature?  It would be nice to do this in a more explicit way, but I suppose that might be difficult

Reply:
Yeah, it's possible to do this more cleanly using `zip --show-files` to first detect whether there is a signature file; I wrestled with my conscience before going with the dirty answer. I'll do the explicit version.